### PR TITLE
fix false positive on links like a..b

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers fixes `a..b` being a link

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits e2dc6c4fcb0b...ba3598f26314 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1484942528672-e2dc6c4fcb0b-ba3598f26314-linux/Markdown-links-e2dc6c4fcb0b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1484942528672-e2dc6c4fcb0b-ba3598f26314-linux/Markdown-links-ba3598f26314.png)  
   Markdown-links comparison
</summary>

File: shared/markdown/parser.pegjs [hunk 1]
```diff
@@ -1,8 +1,15 @@
 {
 	const linkExp = new RegExp(/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)\b/, 'gi')
+    // Instead of encoding all the bad cases into a more complicated regexp lets just add some simple code here
```
Comment (chromakode) [inline shared/markdown/parser.pegjs]: :sheep: Nit: inconsistent indentation

File: shared/markdown/parser.pegjs [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+    function goodLink (link) {
+        if (link.includes('..'))
```
Comment (chromakode) [inline shared/markdown/parser.pegjs]: Isn't ".." a valid occurrence in URL paths?

Comment (chrisnojima) [inline shared/markdown/parser.pegjs]: i'm not trying to make this 100% perfect at this point. its rare you have a .. in a url as a url never needs one of those

**Comment by chrisnojima** [comment]
@chromakode updated the spacing, i have other PRs that expand the link handling after this

**Comment by keybase-ci-visdiff** [comment]
The commits e2dc6c4fcb0b...6f1fe06d208a introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485184784600-e2dc6c4fcb0b-6f1fe06d208a-linux/Markdown-links-e2dc6c4fcb0b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485184784600-e2dc6c4fcb0b-6f1fe06d208a-linux/Markdown-links-6f1fe06d208a.png)  
   Markdown-links comparison
</summary>

File: shared/markdown/parser.pegjs [hunk 3]
```diff
@@ -1,25 +1,33 @@
...
+  // Note: We aren't trying to be 100% perfect here, just getting something that works pretty good and pretty quickly
+  function goodLink (link) {
+    if (link.includes('..'))
+      return false
+        return true
+  }
+
+  function convertLink (text) {
+    const matches = text.match(linkExp)
+      if (matches && goodLink(matches[0])) {
```
Comment (chromakode) [inline shared/markdown/parser.pegjs]: :sheep: Unnecessary indentation here

File: shared/markdown/parser.pegjs [hunk 4]
```diff
@@ -1,25 +1,33 @@
...
-			}
-		} else {
-			return text
-		}
-	}
+  const linkExp = new RegExp(/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)\b/, 'gi')
+  // Instead of encoding all the bad cases into a more complicated regexp lets just add some simple code here
+  // Note: We aren't trying to be 100% perfect here, just getting something that works pretty good and pretty quickly
+  function goodLink (link) {
+    if (link.includes('..'))
```
Comment (chromakode) [inline shared/markdown/parser.pegjs]: `return link.includes('..')`

Comment (chrisnojima) [inline shared/markdown/parser.pegjs]: This isn't going to be the only check so i'm structure the code as such

File: shared/markdown/parser.pegjs (hunk 3 above)
Comment (chrisnojima) [inline shared/markdown/parser.pegjs]: weird, my editor is getting very confused by the pegjs file

**Comment by keybase-ci-visdiff** [comment]
The commits e2dc6c4fcb0b...a297a9b062f8 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485201451214-e2dc6c4fcb0b-a297a9b062f8-linux/Markdown-links-e2dc6c4fcb0b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485201451214-e2dc6c4fcb0b-a297a9b062f8-linux/Markdown-links-a297a9b062f8.png)  
   Markdown-links comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits e2dc6c4fcb0b...9d6c684f685a introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485205832087-e2dc6c4fcb0b-9d6c684f685a-linux/Markdown-links-e2dc6c4fcb0b.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485205832087-e2dc6c4fcb0b-9d6c684f685a-linux/Markdown-links-9d6c684f685a.png)  
   Markdown-links comparison
</summary>

**Pull request merged**